pkg/controllers/dnspolicy/dnspolicy_healthchecks.go

```diff
@@ -99,60 +100,54 @@ func (r *DNSPolicyReconciler) expectedProbesForGateway(ctx context.Context, gw c
-		return nil, nil
+		return healthChecks, nil
 	}
 
-	for _, listener := range gw.Spec.Listeners {
+	for _, address := range gw.Status.Addresses {
+		port := dnsPolicy.Spec.HealthCheck.Port
+		if port == nil {
+			port = &defaultPort
+		}
 
-		log.V(3).Info("getting dnsrecord", "name", dnsRecordName(gw.Name, string(listener.Name)))
-		dnsRecord := &v1alpha1.DNSRecord{}
-		if err := r.Client().Get(ctx, client.ObjectKey{Name: dnsRecordName(gw.Name, string(listener.Name)), Namespace: dnsPolicy.Namespace}, dnsRecord); err != nil {
-			if errors.IsNotFound(err) {
-				continue
-			} else {
-				return nil, err
-			}
+		matches := strings.Split(address.Value, "/")
+		if len(matches) != 2 {
+			return nil, fmt.Errorf(fmt.Sprintf("unable to extract address from address %s ", address.Value))
 		}
-		for _, endpoint := range dnsRecord.Spec.Endpoints {
 
-			log.V(1).Info("reconcileHealthChecks: found DNS Record endpoint", "endpoint", endpoint)
-			if endpoint.RecordType == string(v1alpha1.CNAMERecordType) {
-				// if the CNAME is a subdomain of this dnsrecord, skip it
-				if dns.IsSubDomain(dnsRecord.Name, endpoint.Targets[0]) {
-					continue
-				}
-			} else if endpoint.RecordType != string(v1alpha1.ARecordType) {
-				log.V(1).Info("reconcileHealthChecks: not an A record, skipping")
+		for _, listener := range gw.Spec.Listeners {
+			if strings.Contains(string(*listener.Hostname), "*") {
 				continue
 			}
-			for _, target := range endpoint.Targets {
-				port := dnsPolicy.Spec.HealthCheck.Port
-				if port == nil {
-					port = &defaultPort
-				}
-				log.V(1).Info("reconcileHealthChecks: adding health check for target", "target", target)
-				healthCheck := &v1alpha1.DNSHealthCheckProbe{
-					ObjectMeta: metav1.ObjectMeta{
-						Name:      fmt.Sprintf("%s-%s", target, dnsRecord.Name),
-						Namespace: gw.Namespace,
-						Labels:    commonDNSRecordLabels(client.ObjectKeyFromObject(gw), client.ObjectKeyFromObject(dnsPolicy)),
-					},
-					Spec: v1alpha1.DNSHealthCheckProbeSpec{
-						Port:                     *port,
-						Host:                     dnsRecord.Name,
-						Address:                  target,
-						Path:                     dnsPolicy.Spec.HealthCheck.Endpoint,
-						Protocol:                 *dnsPolicy.Spec.HealthCheck.Protocol,
-						Interval:                 metav1.Duration{Duration: 60 * time.Second},
-						AdditionalHeadersRef:     dnsPolicy.Spec.HealthCheck.AdditionalHeadersRef,
-						FailureThreshold:         dnsPolicy.Spec.HealthCheck.FailureThreshold,
-						ExpectedResponses:        dnsPolicy.Spec.HealthCheck.ExpectedResponses,
-						AllowInsecureCertificate: dnsPolicy.Spec.HealthCheck.AllowInsecureCertificates,
-					},
-				}
-				healthChecks = append(healthChecks, withDNSRecord(dnsRecord, healthCheck))
+			// handle protocol being nil
+			var protocol string
+			if dnsPolicy.Spec.HealthCheck.Protocol == nil {
+				protocol = string(v1alpha1.HttpProtocol)
```

Comment:
Does this make sense as a default? if the value isn't set we end up with a panic. @philbrookes

Reply:
hmm, I think HTTPS probably makes more sense for the default  value here, I imagine that is much more common.

Reply:
Cool. I'll update it. 👍

Reply:
Could the protocol not be determined by the listener?

Reply:
Yes, true. I updated that and also for the default port, we can take that from the listener as well if that makes sense?